File: AccuPay/Repositories/JobLevelRepository.vb
Option Strict On

Imports System.Threading.Tasks
Imports AccuPay.Entity
Imports AccuPay.JobLevels
Imports Microsoft.EntityFrameworkCore

Namespace Global.AccuPay.Repository

    Public Class JobLevelRepository

        Public Async Function GetAllAsync() As Task(Of List(Of JobLevel))

            Using context As New PayrollContext

                Return Await context.JobLevels.
                            Where(Function(j) Nullable.Equals(j.OrganizationID, z_OrganizationID)).
                            ToListAsync

            End Using

        End Function

    End Class

End Namespace

File: AccuPay.Infrastructure/Data/Repositories/JobLevelRepository.cs
using AccuPay.Core.Entities;
using AccuPay.Core.Interfaces;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AccuPay.Infrastructure.Data
{
    public class JobLevelRepository : IJobLevelRepository
    {
        private readonly PayrollContext _context;

        public JobLevelRepository(PayrollContext context)
        {
            _context = context;
        }

        public void Delete(JobLevel jobLevel)
        {
            _context.JobLevels.Remove(jobLevel);
            _context.SaveChanges();
        }

        public IEnumerable<JobLevel> GetAll(int organizationId)
        {
            return _context.JobLevels.
                            Where(x => x.OrganizationID == organizationId).
                            ToList();
        }

        public async Task<IEnumerable<JobLevel>> GetAllAsync(int organizationId)
        {
            return await _context.JobLevels.
                                Where(x => x.OrganizationID == organizationId).
                                ToListAsync();
        }
    }
}
